File: 06Adapter/Students/2019/KucmanBłażej/AdapterPattern/AdapterPattern/ex1/Seabird.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AdapterPattern.ex1
{
    public class Seabird : ISeacraft, IAircraft
    {
        private readonly IAircraft aircraft;
        int height = 0;

        public Seabird(IAircraft aircraft)
        {
            this.aircraft = aircraft;
        }

        // A two-way adapter hides and routes the Target's methods
        // Use Seacraft instructions to implement this one
        public void TakeOff()
        {
            aircraft.TakeOff();
            while (!Airborne)
                IncreaseRevs();
        }

        // Routes this straight back to the Aircraft
        public int Height
        {
            get { return aircraft.Height; }
            set { aircraft.Height = value; }
        }

        // This method is common to both Target and Adaptee
        public  void IncreaseRevs()
        {
            Speed += 10;
            if (Speed > 40)
                aircraft.Height += 100;
        }

        public bool Airborne
        {
            get { return aircraft.Height > 50; }
        }

        public int Speed { get; set; }
    }
}

File: 06Adapter/Students/2019/KawaPiotr/src/main/kotlin/pkawa/zad1/Seabird.kt
package pkawa.zad1

class Seabird : Seacraft(), IAircraft {
    override val airborne: Boolean
        get() = height > 50

    override var height: Int = 0
        private set

    // A two-way adapter hides and routes the Target's methods
    // Use Seacraft instructions to implement this one
    override fun takeOff() {
        while (!airborne) {
            increaseRevs()
        }
    }

    // This method is common to both Target and Adaptee
    override fun increaseRevs() {
        super.increaseRevs()
        if (speed > 40) {
            height += 100
        }
    }
}